test/metabase/events/revision_test.clj

```diff
@@ -146,57 +136,54 @@
                                    :cards [(assoc (apply dissoc dashcard @#'dashboard/excluded-columns-for-dashcard-revision) :series [])])
               :is_reversion false
               :is_creation  false}
-             (mt/derecordize
-              (t2/select-one [Revision :model :model_id :user_id :object :is_reversion :is_creation]
-                :model    "Dashboard"
-                :model_id dashboard-id)))))))
+             (t2/select-one [Revision :model :model_id :user_id :object :is_reversion :is_creation]
+                            :model    "Dashboard"
+                            :model_id dashboard-id))))))
 
 (deftest dashboard-remove-cards-test
-  (testing ":dashboard-remove-cards"
+  (testing :event/dashboard-remove-cards
     (t2.with-temp/with-temp [Dashboard     {dashboard-id :id, :as dashboard} {}
                              Card          {card-id :id}                     (card-properties)
                              DashboardCard dashcard                          {:card_id card-id, :dashboard_id dashboard-id}]
       (t2/delete! (t2/table-name DashboardCard), :id (:id dashcard))
-      (revision/process-revision-event! {:topic :dashboard-remove-cards
-                                         :item  {:id        dashboard-id
-                                                 :actor_id  (mt/user->id :rasta)
-                                                 :dashcards [dashcard]}})
+      (events/publish-event! :event/dashboard-remove-cards
+                             {:id        dashboard-id
+                              :actor_id  (mt/user->id :rasta)
+                              :dashcards [dashcard]})
       (is (= {:model        "Dashboard"
               :model_id     dashboard-id
               :user_id      (mt/user->id :rasta)
               :object       (assoc (dashboard->revision-object dashboard) :cards [])
               :is_reversion false
               :is_creation  false}
-             (mt/derecordize
-              (t2/select-one [Revision :model :model_id :user_id :object :is_reversion :is_creation]
-                :model    "Dashboard"
-                :model_id dashboard-id)))))))
+             (t2/select-one [Revision :model :model_id :user_id :object :is_reversion :is_creation]
+                            :model    "Dashboard"
+                            :model_id dashboard-id))))))
 
 (deftest dashboard-reposition-cards-test
-  (testing ":dashboard-reposition-cards"
+  (testing :event/dashboard-reposition-cards
     (t2.with-temp/with-temp [Dashboard     {dashboard-id :id, :as dashboard} {}
                              Card          {card-id :id}                     (card-properties)
                              DashboardCard dashcard                          {:card_id card-id, :dashboard_id dashboard-id}]
       (t2/update! DashboardCard (:id dashcard) {:size_x 3})
-      (revision/process-revision-event! {:topic :dashboard-reeposition-cards
```

Comment:
Here's a typo that the new code caught. `:dashboard-reeposition-cards` was never a valid event.